src/ng/http.js

```diff
@@ -141,6 +141,9 @@ function defaultHttpResponseTransform(data, headers) {
         try {
           data = fromJson(tempData);
         } catch (e) {
+          if ((contentType && (contentType.indexOf(APPLICATION_JSON) !== 0))) {
+            return tempData;
```

Comment:
Why return `tempData`? If we decide to return the original response, it should be `data`.